cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -418,7 +419,23 @@ private boolean isDeleteLifecycleEnabled(String bucketName) {
       }
       if (rule.getAction() instanceof BucketInfo.LifecycleRule.DeleteLifecycleAction &&
         rule.getCondition().getDaysSinceCustomTime() > 0) {
-        return true;
+        if (!provisionerContext.getProperties().containsKey(DataprocUtils.ARTIFACTS_COMPUTE_HASH_TIME_BUCKET_DAYS)) {
```

Comment:
If this property is coming from `cdap-site.xml` use `provisionerProperties` instead of `provisionerContext.getProperties()` because it only contains properties from compute profile config.

Reply:
On a second thought, `provisionerProperties` only contains the properties from `cdap-site` which are prefixed with `provisioner.system.properties.<provisioner>`, so even `provisionerProperties` won't help here.

Reply:
Discussed offline, working as expected.